gen-go/client/client.go

```diff
@@ -0,0 +1,212 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Clever/resolve-ip/gen-go/models"
+
+	discovery "github.com/Clever/discovery-go"
+)
+
+var _ = json.Marshal
+var _ = strings.Replace
+var _ = strconv.FormatInt
+var _ = bytes.Compare
+
+// Client is used to make requests to the resolve-ip service.
+type Client struct {
+	basePath    string
+	requestDoer doer
+	transport   *http.Transport
+	timeout     time.Duration
+	// Keep the retry doer around so that we can set the number of retries
+	retryDoer      *retryDoer
+	defaultTimeout time.Duration
+}
+
+// New creates a new client. The base path and http transport are configurable.
+func New(basePath string) *Client {
+	base := baseDoer{}
+	tracing := tracingDoer{d: base}
+	retry := retryDoer{d: tracing, defaultRetries: 1}
+
+	return &Client{requestDoer: &retry, retryDoer: &retry, defaultTimeout: 10 * time.Second,
+		transport: &http.Transport{}, basePath: basePath}
+}
+
+// NewFromDiscovery creates a client from the discovery environment variables. This method requires
+// the three env vars: SERVICE_RESOLVE_IP_HTTP_(HOST/PORT/PROTO) to be set. Otherwise it returns an error.
+func NewFromDiscovery() (*Client, error) {
+	url, err := discovery.URL("resolve-ip", "http")
+	if err != nil {
+		return nil, err
+	}
+	return New(url), nil
+}
+
+// WithRetries returns a new client that retries all GET operations until they either succeed or fail the
+// number of times specified.
+func (c *Client) WithRetries(retries int) *Client {
+	c.retryDoer.defaultRetries = retries
```

Comment:
This and WithTimeout are modifying the client state instead of making a new client as documented so if you try to have multiple timeout or retry policies it will silently do the wrong thing